src/App.tsx

```diff
@@ -1,39 +1,58 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
 
-        {' time is '}
+  setTimerForClockName = window.setInterval(() => {
+    this.setState({ clockName: getRandomName() });
+  }, 3300);
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+    });
+    document.addEventListener('contextmenu', this.handleRemoveClock);
+    document.addEventListener('click', this.handleAddClock);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleRemoveClock);
+    document.removeEventListener('click', this.handleAddClock);
+  }
+
+  handleRemoveClock = () => {
+    this.setState({ hasClock: false });
+  };
```

Comment:
you can handle event here
```suggestion
  handleRemoveClock = (event) => {
    event.preventDefault();
    this.setState({ hasClock: false });
  };
```